src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java

```diff
@@ -253,20 +243,18 @@ public static boolean isOuterMostType(DetailAST node) {
     public static boolean isLocalVariableDef(DetailAST node) {
         boolean localVariableDef = false;
         // variable declaration?
-        if (node.getType() == TokenTypes.VARIABLE_DEF) {
+        if (TokenUtil.isOfType(node, TokenTypes.VARIABLE_DEF)) {
             final DetailAST parent = node.getParent();
-            final int type = parent.getType();
-            localVariableDef = type == TokenTypes.SLIST
-                    || type == TokenTypes.FOR_INIT
-                    || type == TokenTypes.FOR_EACH_CLAUSE;
+            localVariableDef = TokenUtil.isOfType(parent, TokenTypes.SLIST,
+                                TokenTypes.FOR_INIT, TokenTypes.FOR_EACH_CLAUSE);
         }
         // catch parameter?
-        if (node.getType() == TokenTypes.PARAMETER_DEF) {
+        if (TokenUtil.isOfType(node, TokenTypes.PARAMETER_DEF)) {
             final DetailAST parent = node.getParent();
-            localVariableDef = parent.getType() == TokenTypes.LITERAL_CATCH;
+            localVariableDef = TokenUtil.isOfType(parent.getType(), TokenTypes.LITERAL_CATCH);
         }
 
-        if (node.getType() == TokenTypes.RESOURCE) {
+        if (TokenUtil.isOfType(node, TokenTypes.RESOURCE)) {
```

Comment:
please restore single node verification to use `==`, it is more concise and shorter and performance is better.

Reply:
done